java/ql/src/experimental/CWE-532/SensitiveInfoLog.ql

```diff
@@ -0,0 +1,64 @@
+/**
+ * @id java/sensitiveinfo-in-logfile
+ * @name Insertion of sensitive information into log files
+ * @description Writing sensitive information to log files can give valuable guidance to an attacker or expose sensitive user information.
+ * @kind problem
+ * @tags security
+ *       external/cwe-532
+ */
+
+import java
+import semmle.code.java.dataflow.TaintTracking
+import DataFlow
+import PathGraph
+
+/**
+ * Gets a regular expression for matching names of variables that indicate the value being held is a credential
+ */
+private string getACredentialRegex() {
+  result = "(?i).*pass(wd|word|code|phrase)(?!.*question).*" or
+  result = "(?i)(.*username|url).*"
+}
+
+/** The variable or concatenated string with the variable that keeps sensitive information judging by its name * */
+class CredentialExpr extends Expr {
+  CredentialExpr() {
+    exists(Variable v |
+      (this.(AddExpr).getAnOperand() = v.getAnAccess() or this = v.getAnAccess())
```

Comment:
This can be simplified to just `this = v.getAnAccess()`, as the `AddExpr` step will be included by the `TaintTracking::localTaintStep(node1, node2)` you added.

Reply:
Thanks again @aschackmull . I've simplified the CredentialExpr and committed the code.